sound/soc/sof/intel/hda.c

```diff
@@ -681,6 +681,19 @@ void hda_ipc4_dump(struct snd_sof_dev *sdev)
 		hipci, hipcie, hipct, hipcte, hipcctl);
 }
 
+bool hda_ipc4_can_send_message(struct snd_sof_dev *sdev)
+{
+	struct sof_intel_hda_dev *hda = sdev->pdata->hw_pdata;
+	const struct sof_intel_dsp_desc *chip = hda->desc;
+	u32 val;
+
+	val = snd_sof_dsp_read(sdev, HDA_DSP_BAR, chip->ipc_req);
+	if (val & chip->ipc_req_mask)
+		return false;
+
+	return true;
```

Comment:
this function is always called as `!hda_ipc4_can_send_message()` so you could invert its meaning like `hda_ipc4_busy()` and revert here to `return !!(val & chip->ipc_req_mask)` or even `return !!(chip->ipc_req_mask & snd_sof_dsp_read(sdev, HDA_DSP_BAR, chip->ipc_req))` but that's probably too long. But this is absolutely optional of course

Reply:
I was trying to find a descriptive function name for checking that the downstream communication channel is busy, but all came back too long, so ended up with a positive one.
Probably `hda_ipc4_send_channel_is_busy()` would work better?

I have played with the idea to manage the sdev->ipc->sending_delayed flag in there as well, but that is just overloading a function with a feature you would not have guessed from it's name...

Reply:
Or `hda_ipc4_tx_channel_is_busy()`